parameter/ParameterHandle.cpp

```diff
@@ -74,407 +74,190 @@ string CParameterHandle::getKind() const
+template <class T>
+bool CParameterHandle::getAs(T &value, string &error) const
 {
-    // Check operation validity
-    if (!checkGetValidity(false, strError)) {
+    if (not checkGetValidity(false, error)) {
 
         return false;
     }
+
     // Ensure we're safe against blackboard foreign access
     lock_guard<mutex> autoLock(_pParameterMgr->getBlackboardMutex());
 
-    // Define access context
-    CParameterAccessContext parameterAccessContext(strError, _pParameterMgr->getParameterBlackboard());
+    CParameterAccessContext parameterAccessContext(error, _pParameterMgr->getParameterBlackboard());
 
-    return _pBaseParameter->accessAsBoolean(bValue, false, parameterAccessContext);
+    return _pBaseParameter->access(value, false, parameterAccessContext);
 }
 
-bool CParameterHandle::setAsBooleanArray(const std::vector<bool>& abValues, string& strError)
+template <class T>
+bool CParameterHandle::setAsArrayOf(const std::vector<T>& values, string &error) const
 {
     // Check operation validity
-    if (!checkSetValidity(abValues.size(), strError)) {
+    if (not checkSetValidity(values.size(), error)) {
 
         return false;
     }
+
     // Ensure we're safe against blackboard foreign access
     lock_guard<mutex> autoLock(_pParameterMgr->getBlackboardMutex());
 
-    // When in tuning mode, silently skip the request
+    // When in tuning mode, silently skip "set" requests
     if (_pParameterMgr->tuningModeOn()) {
-
         return true;
     }
 
-    // Define access context
-    CParameterAccessContext parameterAccessContext(strError, _pParameterMgr->getParameterBlackboard());
+    CParameterAccessContext parameterAccessContext(error, _pParameterMgr->getParameterBlackboard());
 
     // Copy values for type adaptation
-    std::vector<bool> abUserValues = abValues;
-
-    return _pBaseParameter->accessAsBooleanArray(abUserValues, true, parameterAccessContext);
+    std::vector<T> auiUserValues = values;
+    return _pBaseParameter->access(auiUserValues, true, parameterAccessContext);
 }
 
-bool CParameterHandle::getAsBooleanArray(std::vector<bool>& abValues, string& strError) const
+template <class T>
+bool CParameterHandle::getAsArrayOf(std::vector<T>& values, string &error) const
 {
-    // Check operation validity
-    if (!checkGetValidity(true, strError)) {
+    if (not checkGetValidity(true, error)) {
 
         return false;
     }
+
     // Ensure we're safe against blackboard foreign access
     lock_guard<mutex> autoLock(_pParameterMgr->getBlackboardMutex());
 
-    // Define access context
-    CParameterAccessContext parameterAccessContext(strError, _pParameterMgr->getParameterBlackboard());
+    CParameterAccessContext parameterAccessContext(error, _pParameterMgr->getParameterBlackboard());
 
-    return _pBaseParameter->accessAsBooleanArray(abValues, false, parameterAccessContext);
+    return _pBaseParameter->access(values, true, parameterAccessContext);
```

Comment:
Is this `true` the only difference between `getAsArrayOf` and `getAs` ? If so, consider merging the two.